core/heap.go

```diff
@@ -0,0 +1,55 @@
+package core
```

Comment:
It's standard Go library. They already wrote unit tests for this package. [code](https://github.com/golang/go/blob/master/src/container/heap/heap_test.go)

Do you want to add that test file in our project?